packages/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/MarkersController.java

```diff
@@ -0,0 +1,115 @@
+package io.flutter.plugins.googlemaps;
+
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.model.Marker;
+import com.google.android.gms.maps.model.MarkerOptions;
+import io.flutter.plugin.common.MethodChannel;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+class MarkersController {
+
+  private final Map<MarkerId, MarkerController> markerIdToController;
+  private final Map<String, MarkerId> googleMapsMarkerIdToDartMarkerId;
+  private final MethodChannel methodChannel;
+  private GoogleMap googleMap;
+
+  MarkersController(MethodChannel methodChannel) {
+    this.markerIdToController = new HashMap<>();
+    this.googleMapsMarkerIdToDartMarkerId = new HashMap<>();
+    this.methodChannel = methodChannel;
+  }
+
+  void setGoogleMap(GoogleMap googleMap) {
+    this.googleMap = googleMap;
+  }
+
+  void addMarkers(List<Object> markersToAdd) {
+    if (markersToAdd != null) {
+      for (Object markerToAdd : markersToAdd) {
+        addMarker(markerToAdd);
+      }
+    }
+  }
+
+  void changeMarkers(List<Object> markersToChange) {
+    if (markersToChange != null) {
+      for (Object markerToChange : markersToChange) {
+        changeMarker(markerToChange);
+      }
+    }
+  }
+
+  void removeMarkers(List<Object> markerIdsToRemove) {
+    if (markerIdsToRemove == null) {
+      return;
+    }
+    for (Object rawMarkerId : markerIdsToRemove) {
+      if (rawMarkerId != null) {
+        MarkerId markerId = new MarkerId((String) rawMarkerId);
+        final MarkerController markerController = markerIdToController.remove(markerId);
+        if (markerController != null) {
+          markerController.remove();
+          googleMapsMarkerIdToDartMarkerId.remove(markerController.getGoogleMapsMarkerId());
+        }
+      }
+    }
+  }
+
+  boolean onMarkerTap(String googleMarkerId) {
+    MarkerId markerId = googleMapsMarkerIdToDartMarkerId.get(googleMarkerId);
+    if (markerId == null) {
+      return false;
+    }
+    methodChannel.invokeMethod("marker#onTap", Convert.toJson(markerId));
+    MarkerController markerController = markerIdToController.get(markerId);
+    if (markerController != null) {
+      return markerController.consumeTapEvents();
```

Comment:
Why do we need to do this here?

Reply:
The method contract is:
``` true if the listener has consumed the event (i.e., the default behavior should not occur); false otherwise (i.e., the default behavior should occur). The default behavior is for the camera to move to the marker and an info window to appear.
```

https://developers.google.com/android/reference/com/google/android/gms/maps/GoogleMap.OnMarkerClickListener.html#onMarkerClick(com.google.android.gms.maps.model.Marker)